src/CowAmmModule.sol

```diff
@@ -0,0 +1,299 @@
+     */
+    constructor(
+        GPv2Settlement _settler,
+        ExtensibleFallbackHandler _extensibleFallbackHandler,
+        ComposableCoW _composableCow,
+        IConditionalOrder _handler
+    ) {
+        // GPv2 specifics to make sure we set the right things and they're immutable!
+        SETTLER = _settler;
+        COW_DOMAIN_SEPARATOR = SETTLER.domainSeparator();
+        VAULT_RELAYER = address(SETTLER.vaultRelayer());
+
+        // ComposableCoW specifics
+        EXTENSIBLE_FALLBACK_HANDLER = _extensibleFallbackHandler;
+        COMPOSABLE_COW = _composableCow;
+        HANDLER = _handler;
+    }
+
+    /**
+     * @notice Creates a new CoW AMM with the given parameters.
+     * @param token0 The address of the first token in the pair.
+     * @param token1 The address of the second token in the pair.
+     * @param minTradedToken0 The minimum amount of token0 before the AMM attempts auto-rebalance.
+     * @param priceOracle The address of the price oracle to use for the AMM.
+     * @param priceOracleData The data to pass to the price oracle.
+     * @param appData The app data to pass to the AMM.
+     */
+    function createAmm(
+        IERC20 token0,
+        IERC20 token1,
+        uint256 minTradedToken0,
+        address priceOracle,
+        bytes calldata priceOracleData,
+        bytes32 appData
+    ) external {
+        // Assume the caller is a Safe
+        Safe safe = Safe(payable(msg.sender));
+
+        if (activeOrders[safe] != EMPTY_AMM_HASH) {
+            // If there is an active order, the user must close it before creating a new one
+            // This is not fool-proof as the user could create another AMM outside of this module
+            revert ActiveAMM();
+        }
+
+        _createAmm(safe, token0, token1, minTradedToken0, priceOracle, priceOracleData, appData);
+    }
+
+    /**
+     * @notice Replaces the active CoW AMM with a new one.
+     * @param token0 The address of the first token in the pair.
+     * @param token1 The address of the second token in the pair.
+     * @param minTradedToken0 The minimum amount of token0 before the AMM attempts auto-rebalance.
+     * @param priceOracle The address of the price oracle to use for the AMM.
+     * @param priceOracleData The data to pass to the price oracle.
+     * @param appData The app data to pass to the AMM.
+     * @dev This function internally just calls `closeAmm` and then `createAmm`.
+     */
+    function replaceAmm(
+        IERC20 token0,
+        IERC20 token1,
+        uint256 minTradedToken0,
+        address priceOracle,
+        bytes calldata priceOracleData,
+        bytes32 appData
+    ) external {
+        // Assume the caller is a Safe
+        Safe safe = Safe(payable(msg.sender));
+
+        bytes32 _activeOrder = activeOrders[safe];
+        if (_activeOrder != EMPTY_AMM_HASH) {
+            _closeAmm(safe, _activeOrder);
+        }
```

Comment:
In line with the comment above: here we could revert if the user tries to replace an order and there is no order to replace.

```suggestion
        if (_activeOrder == EMPTY_AMM_HASH) {
            revert NoActiveOrderToReplace();
        }
        _closeAmm(safe, _activeOrder);
```